src/edu/ucsb/nceas/metacat/admin/upgrade/HashStoreUpgrader.java

```diff
@@ -391,14 +342,16 @@ private void handleErrorFile(File file, StringBuffer buffer) {
         }
     }
 
-    private void convert(Path path, String finalId, InputStream sysMetaInput, String checksum,
+    private void convert(Path path, String finalId, SystemMetadata sysMeta, String checksum,
                          String algorithm, BufferedWriter nonMatchingChecksumWriter,
                          BufferedWriter noSuchAlgorithmWriter,
                          BufferedWriter generalWriter, BufferedWriter savingChecksumTableWriter) {
         ObjectMetadata metadata = null;
         try {
-            metadata =
-                converter.convert(path, finalId, sysMetaInput, checksum, algorithm);
+            try (InputStream sysMetaInput = convertSystemMetadata(sysMeta)) {
```

Comment:
I don't think you need another nested `try` here - you can just add the `(InputStream sysMetaInput = convertSystemMetadata(sysMeta))` to the first `try` statement